days/day-04/solutions/day04.tholok97.go

```diff
@@ -0,0 +1,196 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+// validator is any function that can tell you if the given string is valid
+// according to some criteria
+type validator func(string) bool
+
+// fieldValidators provides a validator for each key
+type fieldValidators map[string]validator
```

Comment:
This is like a generic container. I thought golang did not have generics ?! :smile: You proved me wrong! Why is this not a generic container=? @hypirion help! My world-view is broken

Reply:
Go _has_ parametric polymorphism (aka generics), but it's only for [builtin](https://godoc.org/builtin) types and functions.

So yeah, the `map` type here is a generic, but `fieldValidators` or any of the other types are most certainly not generic.

Reply:
Aww. I see :+1: Thank you for the explanation